### ISSUE TITLE
Retaining results from discarded window portion.

### ISSUE THREAD

**Issue by DillonJ** [Spine-project/SpineOpt.jl#410]
Consider the following structure :

![image](https://user-images.githubusercontent.com/7080191/137488537-a14388f3-073e-4480-ae6e-50090f2d40fb.png)

Related to #409, we may wish to keep the results from the otherwise discarded portion of the window as they could represent, for example, day-ahead or intraday prices that we want to keep. 

In the example above, in window 1, we temporarily keep the results from the whole window but the window 1 results from TBs 2 and 3 will get overwritten by the new window 2 results. We want a way to keep them.

A solution is a method parameter on `output__report` that specifies whether the overlapping window result is kept or not. It could be called `window_output_method` with values:  `entire_window` or `non_overlapping_only` and the default would be `non_overlapping_only`.

The advantage of this solution is that it can be changed per output. We might not want all outputs kept.

An implication of this is that for a particular timeslice, we can have multiple results, depending on the look-ahead - so we need a new index that corresponds to the window. The most obvious and consistent would be to use analysis_time corresponding to the window start.

**Comment by manuelma** [comment]
In this case, I think it makes more sense to have a boolean parameter called something like `overwrite_results_on_rolling`. To me it's much more clear (I have difficulties understanding the meaning of `non_overlapping_only` for instance).

But I fully like the idea. As in #409, I only disagree on the parameter name.

**Comment by DillonJ** [comment]
I agree that `overwrite_results_on_rolling` is more understandable - it's default value would need to be true

**Comment by DillonJ** [comment]
Hi folks,

Returning to this - the original intention of this issue was to provide functionality so we could have access to results from the overlapping portion of suquential windows. The solution was `overwrite_results_on_rolling` where the full window is always written but we have the choice of whether or not to overwrite the overlapping time indices or not.

I believe this implements the desired functionality but I wonder if it does so efficiently. For example, I have a stochastic model where the full window is 48h long but it rolls 6h at a time. The model consists of a 6h realisation stage and at 42h stocahstic stage with 6 scenarios.  I don't want to retain results for the overlapping 42h stochastic stage and I feel like there must be a very large overhead in writing this data each time only for it to get overwritten on the next window. Surely there would be performance advantage in not writing these results in the first place, as thay will be overwritten?

What do you think @manuelma ?

**Comment by manuelma** [comment]
I think you're right @DillonJ - I just tested it in your model above, and croping outputs to the current window has a decent impact. Saving outputs goes from 1.7 secs to 0.45 secs per window which means a few minutes in a long run. Will test that everything goes right and then push to master.

Although, all the croping I did was for saving **outputs**. This is different to saving **values** for fixing the history in the next window, but I guess we could also crop in this latter case. For example, in the model you @DillonJ describe, we would just need to save values for hours 1 to 12, because everything from hour 13 to 48 will never be used, am I right?
